Investigate javadoc WARNINGS when executing `mvn javadoc:javadoc`

This is a follow up to #164 

The current set of warnings to investigate are:

```
9 warnings
[WARNING] Javadoc Warnings
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\ConditionalTagSupport.java:55: warning: no @throws for jakarta.servlet.jsp.JspTagException
[WARNING] protected abstract boolean condition() throws JspTagException;
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\IndexedValueExpression.java:37: warning: no @param for orig
[WARNING] public IndexedValueExpression(ValueExpression orig, int i) {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\IndexedValueExpression.java:37: warning: no @param for i
[WARNING] public IndexedValueExpression(ValueExpression orig, int i) {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\IteratedExpression.java:55: warning: no @return
[WARNING] public Object getItem(ELContext context, int i) {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:185: warning: no description for @exception
[WARNING] * @exception jakarta.servlet.jsp.JspTagException
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:196: warning: no description for @exception
[WARNING] * @exception jakarta.servlet.jsp.JspTagException
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:453: warning: no @throws for jakarta.servlet.jsp.JspTagException
[WARNING] protected void validateBegin() throws JspTagException {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:462: warning: no @throws for jakarta.servlet.jsp.JspTagException
[WARNING] protected void validateEnd() throws JspTagException {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:471: warning: no @throws for jakarta.servlet.jsp.JspTagException
[WARNING] protected void validateStep() throws JspTagException {
[WARNING] ^
```

I'm getting a new warning when running the javadoc command: 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:javadoc (default-cli) on project jakarta.servlet.jsp.jstl-api: An error has occurred in Javadoc report generation: 
[ERROR] Exit code: 1 - error: module not found: jakarta.servlet.jsp.jstl
```

I tried on the jsp-api and got a similar error.

> I'm getting a new warning when running the javadoc command:
> 
> ```
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:javadoc (default-cli) on project jakarta.servlet.jsp.jstl-api: An error has occurred in Javadoc report generation: 
> [ERROR] Exit code: 1 - error: module not found: jakarta.servlet.jsp.jstl
> ```
> 
> I tried on the jsp-api and got a similar error.
> 
> Could be something with my JDK....? `OpenJDK Runtime Environment Temurin-11.0.13+8 (build 11.0.13+8)`

@volosied I get the same error, if I remove the `module-info.java` then it runs fine. So it must have something to do with the `module-info.java` that was added.

This happens if you are running the javadoc:javadoc phase from a clean build

```bash
└> rm -rf target 
✏️ [starksm@Scotts-iMacPro#10768]-(api)-[git://master ✔]-
└> mvn -Pstaging javadoc:javadoc
[INFO] Scanning for projects...
[INFO] 
[INFO] -------< jakarta.servlet.jsp.jstl:jakarta.servlet.jsp.jstl-api >--------
[INFO] Building Jakarta Standard Tag Library API 3.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] >>> maven-javadoc-plugin:3.3.1:javadoc (default-cli) > generate-sources @ jakarta.servlet.jsp.jstl-api >>>
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce-maven) @ jakarta.servlet.jsp.jstl-api ---
[INFO] 
[INFO] --- spec-version-maven-plugin:2.1:set-spec-properties (default) @ jakarta.servlet.jsp.jstl-api ---
[INFO] 
[INFO] -- spec properties --
[INFO] spec.bundle.symbolic-name = jakarta.servlet.jsp.jstl-api
[INFO] spec.implementation.version = 3.0.0
[INFO] spec.extension.name = jakarta.servlet.jsp.jstl
[INFO] spec.bundle.spec.version = 3.0
[INFO] spec.specification.version = 3.0
[INFO] spec.bundle.version = 3.0.0
[INFO] 
[INFO] 
[INFO] --- build-helper-maven-plugin:3.2.0:timestamp-property (timestamp-property) @ jakarta.servlet.jsp.jstl-api ---
[INFO] 
[INFO] <<< maven-javadoc-plugin:3.3.1:javadoc (default-cli) < generate-sources @ jakarta.servlet.jsp.jstl-api <<<
[INFO] 
[INFO] 
[INFO] --- maven-javadoc-plugin:3.3.1:javadoc (default-cli) @ jakarta.servlet.jsp.jstl-api ---
[ERROR] Creating an aggregated report for both named and unnamed modules is not possible.
[ERROR] Ensure that every module has a module descriptor or is a jar with a MANIFEST.MF containing an Automatic-Module-Name.
[ERROR] Fix the following projects:
[ERROR]  - jakarta.servlet.jsp.jstl:jakarta.servlet.jsp.jstl-api
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.233 s
[INFO] Finished at: 2022-01-17T12:02:41-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.3.1:javadoc (default-cli) on project jakarta.servlet.jsp.jstl-api: An error has occurred in Javadoc report generation: Aggregator report contains named and unnamed modules -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

If you run javadoc:javadoc phase after an install or package you don't see the module problem.

The 9 warnings are legitimate missing javadoc tags that I have a fix for in a PR that I will submit today.

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/ConditionalTagSupport.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/ConditionalTagSupport.java
@@ -52,6 +52,7 @@ public abstract class ConditionalTagSupport extends TagSupport {
      * per tag invocation by <code>doStartTag()</code>.
      *
      * @return a boolean representing the condition that a particular subclass uses to drive its conditional logic.
+     * @throws JspTagException - on failure
      */
     protected abstract boolean condition() throws JspTagException;
 

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/ConditionalTagSupport.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/ConditionalTagSupport.java
@@ -52,7 +52,7 @@ public abstract class ConditionalTagSupport extends TagSupport {
      * per tag invocation by <code>doStartTag()</code>.
      *
      * @return a boolean representing the condition that a particular subclass uses to drive its conditional logic.
-     * @throws JspTagException - on failure
+     * @throws JspTagException on failure
      */
     protected abstract boolean condition() throws JspTagException;
 

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/IndexedValueExpression.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/IndexedValueExpression.java
@@ -32,7 +32,8 @@ public final class IndexedValueExpression extends ValueExpression {
     protected final ValueExpression orig;
 
     /**
-     *
+     * @param orig - ValueExpression that refers to a specific member of an indexed variable
+     * @param i - the index value
      */
     public IndexedValueExpression(ValueExpression orig, int i) {
         this.i = Integer.valueOf(i);

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/IndexedValueExpression.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/IndexedValueExpression.java
@@ -32,8 +32,8 @@ public final class IndexedValueExpression extends ValueExpression {
     protected final ValueExpression orig;
 
     /**
-     * @param orig - ValueExpression that refers to a specific member of an indexed variable
-     * @param i - the index value
+     * @param orig ValueExpression that refers to a specific member of an indexed variable
+     * @param i the index value
      */
     public IndexedValueExpression(ValueExpression orig, int i) {
         this.i = Integer.valueOf(i);

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/IteratedExpression.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/IteratedExpression.java
@@ -41,6 +41,11 @@ public final class IteratedExpression /* implements Serializable */ {
     private int index;
     private Iterator iter;
 
+    /**
+     *
+     * @param orig - ValueExpression to process
+     * @param delims - Delimiters to use to split a String expression
+     */
     public IteratedExpression(ValueExpression orig, String delims) {
         this.orig = orig;
         this.delims = delims;
@@ -51,6 +56,7 @@ public final class IteratedExpression /* implements Serializable */ {
      *
      * @param context The ELContext used to evaluate the ValueExpression
      * @param i The index of the item to be retrieved
+     * @return The value expression result at i
      */
     public Object getItem(ELContext context, int i) {
 

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/IteratedExpression.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/IteratedExpression.java
@@ -43,8 +43,8 @@ public final class IteratedExpression /* implements Serializable */ {
 
     /**
      *
-     * @param orig - ValueExpression to process
-     * @param delims - Delimiters to use to split a String expression
+     * @param orig ValueExpression to process
+     * @param delims Delimiters to use to split a String expression
      */
     public IteratedExpression(ValueExpression orig, String delims) {
         this.orig = orig;

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java
@@ -183,7 +183,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
      * </p>
      *
      * @return <code>true</code> if there is at least one more item to iterate over, <code>false</code> otherwise
-     * @exception jakarta.servlet.jsp.JspTagException
+     * @exception jakarta.servlet.jsp.JspTagException - thrown on error
      * @see #next
      */
     protected abstract boolean hasNext() throws JspTagException;
@@ -194,7 +194,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
      * Subclasses can assume that prepare() will be called once for each invocation of doStartTag() in the superclass.
      * </p>
      *
-     * @exception jakarta.servlet.jsp.JspTagException
+     * @exception jakarta.servlet.jsp.JspTagException - thrown on error
      */
     protected abstract void prepare() throws JspTagException;
 
@@ -450,6 +450,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
 
     /**
      * Ensures the "begin" property is sensible, throwing an exception expected to propagate up if it isn't
+     * @throws JspTagException - thrown on error
      */
     protected void validateBegin() throws JspTagException {
         if (begin < 0) {
@@ -459,6 +460,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
 
     /**
      * Ensures the "end" property is sensible, throwing an exception expected to propagate up if it isn't
+     * @throws JspTagException - thrown on error
      */
     protected void validateEnd() throws JspTagException {
         if (end < 0) {
@@ -468,6 +470,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
 
     /**
      * Ensures the "step" property is sensible, throwing an exception expected to propagate up if it isn't
+     * @throws JspTagException - thrown on error
      */
     protected void validateStep() throws JspTagException {
         if (step < 1) {

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java
@@ -183,7 +183,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
      * </p>
      *
      * @return <code>true</code> if there is at least one more item to iterate over, <code>false</code> otherwise
-     * @exception jakarta.servlet.jsp.JspTagException - thrown on error
+     * @exception jakarta.servlet.jsp.JspTagException thrown on error
      * @see #next
      */
     protected abstract boolean hasNext() throws JspTagException;
@@ -194,7 +194,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
      * Subclasses can assume that prepare() will be called once for each invocation of doStartTag() in the superclass.
      * </p>
      *
-     * @exception jakarta.servlet.jsp.JspTagException - thrown on error
+     * @exception jakarta.servlet.jsp.JspTagException thrown on error
      */
     protected abstract void prepare() throws JspTagException;
 
@@ -460,7 +460,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
 
     /**
      * Ensures the "end" property is sensible, throwing an exception expected to propagate up if it isn't
-     * @throws JspTagException - thrown on error
+     * @throws JspTagException thrown on error
      */
     protected void validateEnd() throws JspTagException {
         if (end < 0) {
@@ -470,7 +470,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
 
     /**
      * Ensures the "step" property is sensible, throwing an exception expected to propagate up if it isn't
-     * @throws JspTagException - thrown on error
+     * @throws JspTagException thrown on error
      */
     protected void validateStep() throws JspTagException {
         if (step < 1) {

--- a/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java
+++ b/api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java
@@ -450,7 +450,7 @@ public abstract class LoopTagSupport extends TagSupport implements LoopTag, Iter
 
     /**
      * Ensures the "begin" property is sensible, throwing an exception expected to propagate up if it isn't
-     * @throws JspTagException - thrown on error
+     * @throws JspTagException thrown on error
      */
     protected void validateBegin() throws JspTagException {
         if (begin < 0) {

